apps/shell/app.js

```diff
@@ -46,16 +46,17 @@ var wss = new WebSocket.Server({ server: server });
 
 wss.on('connection', function connection (ws, req) {
   var match;
-  var host = process.env.DEFAULT_SSHHOST || 'localhost';
+  var host = process.env.DEFAULT_SSHHOST;
   var dir;
   var term;
   var cmd, args;
 
   console.log('Connection established');
 
+  var whitelist = process.env.SSHHOST_WHITELIST.split(':')
```

Comment:
If SSHHOST_WHITELIST is not set, calling split on undefined will raise an exception crashing the app.

Reply:
Whitelist should also be determined and set at the top of app.js instead of during the connection block. So a global variable "whitelist" would be created and used.